packages/@aws-cdk/aws-route53/lib/vpc-endpoint-service-domain-name.ts

```diff
@@ -0,0 +1,197 @@
+      onCreate: enablePrivateDnsAction,
+      onUpdate: enablePrivateDnsAction,
+      onDelete: removePrivateDnsAction,
+      policy: AwsCustomResourcePolicy.fromSdkCalls({
+        resources: AwsCustomResourcePolicy.ANY_RESOURCE,
+      }),
+    });
+
+    // Look up the name/value pair if the domain changes, or the service changes,
+    // which would cause the values to be different. If the unique ID changes,
+    // the resource may be entirely recreated, so we will need to look it up again.
+    const lookup = hashcode(Names.uniqueId(this) + serviceId + privateDnsName);
+
+    // Create the custom resource to look up the name/value pair generated by AWS
+    // after the previous API call
+    const retriveNameValuePairAction = {
+      service: 'EC2',
+      action: 'describeVpcEndpointServiceConfigurations',
+      parameters: {
+        ServiceIds: [serviceId],
+      },
+      physicalResourceId: PhysicalResourceId.of(lookup),
+    };
+    const getNames = new AwsCustomResource(this, 'GetNames', {
+      onCreate: retriveNameValuePairAction,
+      onUpdate: retriveNameValuePairAction,
+      policy: AwsCustomResourcePolicy.fromSdkCalls({
+        resources: AwsCustomResourcePolicy.ANY_RESOURCE,
+      }),
+    });
+
+    // We only want to call and get the name/value pair after we've told AWS to enable Private DNS
+    // If we call before then, we'll get an empty pair of values.
+    getNames.node.addDependency(enable);
+
+    // Get the references to the name/value pair associated with the endpoint service
+    const name = getNames.getResponseField('ServiceConfigurations.0.PrivateDnsNameConfiguration.Name');
+    const value = getNames.getResponseField('ServiceConfigurations.0.PrivateDnsNameConfiguration.Value');
+
+    return { name, value, serviceId };
+  }
+
+  /**
+   * Creates a Route53 entry and a Custom Resource which explicitly tells AWS to verify ownership
+   * of the domain name attached to an endpoint service.
+   */
+  private verifyPrivateDnsConfiguration(config: PrivateDnsConfiguration, publicHostedZone: IPublicHostedZone) {
+    // Create the TXT record in the provided hosted zone
+    const verificationRecord = new TxtRecord(this, 'DnsVerificationRecord', {
+      recordName: config.name,
+      values: [config.value],
+      zone: publicHostedZone,
+    });
+
+    // Tell the endpoint service to verify the domain ownership
+    const startVerificationAction = {
+      service: 'EC2',
+      action: 'startVpcEndpointServicePrivateDnsVerification',
+      parameters: {
+        ServiceId: config.serviceId,
+      },
+      physicalResourceId: PhysicalResourceId.of(Fn.join(':', [config.name, config.value])),
+    };
+    const startVerification = new AwsCustomResource(this, 'StartVerification', {
+      onCreate: startVerificationAction,
+      onUpdate: startVerificationAction,
+      policy: AwsCustomResourcePolicy.fromSdkCalls({
+        resources: AwsCustomResourcePolicy.ANY_RESOURCE,
```

Comment:
Again feels like we should be able to construct an ARN here.

Reply:
Switched to ARN